File: backend/src/xerago_intelligence/types/intelligence.py
"""Read-model types for intelligence API responses."""

from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True)
class DepartmentMappingItem:
    department_name: str
    department_relevance_score: int
    impact_summary: str | None = None
    impact_category: str | None = None
    opportunity_type: str | None = None
    department_opportunity_score: int | None = None
    impact_reason: str | None = None
    impact_version: str | None = None


@dataclass(frozen=True)
class IntelligenceRecord:
    artifact_id: str
    title: str
    url: str
    published_at: datetime
    summary: str
    why_it_matters: str
    domain: str
    signal_type: str
    confidence_score: int
    validation_status: str
    strategic_score: int | None
    priority_level: str | None
    departments: tuple[DepartmentMappingItem, ...]
    department: str | None


@dataclass(frozen=True)
class IntelligencePage:
    items: list[IntelligenceRecord]
    page: int
    page_size: int
    total: int

File: frontend/src/types/intelligence.ts
export type IntelligenceIcon = 'spark' | 'trend' | 'alert'

export type PriorityLevel = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL'

export interface DepartmentMappingItem {
  department_name: string
  department_relevance_score: number
  impact_summary?: string | null
  impact_category?: string | null
  opportunity_type?: string | null
  department_opportunity_score?: number | null
  impact_reason?: string | null
  impact_version?: string | null
}

/** Impact fields for the logged-in department only (dashboard cards). */
export interface DepartmentImpactView {
  impact_summary: string | null
  impact_category: string | null
  opportunity_type: string | null
  department_opportunity_score: number | null
}

export interface IntelligenceItem {
  artifact_id: string
  title: string
  url: string
  published_at: string
  summary: string
  why_it_matters: string
  domain: string
  signal_type: string
  confidence_score: number
  validation_status: string
  strategic_score: number
  priority_level: PriorityLevel
  departments?: DepartmentMappingItem[]
  department?: string | null
}

export interface IntelligenceState {
  loading: boolean
  error: string | null
  data: IntelligenceItem[]
}
